test/test_nestedtensor.py

```diff
@@ -3997,6 +3997,22 @@ def check_size(nt1, nt2, nt3, nt4):
         nt1_t, nt2_t, nt3_t, nt4_t = (x.transpose(1, 2) for x in (nt1, nt2, nt3, nt4))
         check_size(nt1_t, nt2_t, nt3_t, nt4_t)
 
+    @skipCUDAIf(not SM70OrLater, "GPU capability is < SM70")
+    def test_return_nt_constructed_in_graph(self, device):
+        def fn(values, offsets):
+            nt = torch.nested.nested_tensor_from_jagged(values, offsets)
+            nt = nt.cos()
+            return nt
+
+        values = torch.rand((10, 8), device=device)
+        offsets = torch.tensor([0, 1, 3, 6, 10], device=device)
+
+        fn_c = torch.compile(fn, fullgraph=True)
+
+        ref = fn(values, offsets)
+        res = fn_c(values, offsets)
+        self.assertEqual(ref, res)
```

Comment:
It feels like this test should still pass without the changes below because our outer sizes are not symbolic in this case? 
e.g. when we create the nested int in the ctor during tracing, we'd go through get_nested_int and that would return a non-symbolic nested int.

Reply:
> get_nested_int and that would return a non-symbolic nested int.

yes, but it gets a new nested int - so without this change, the outer_size is (4, j2, 8) and the returned tensor has (4, j3, 8)

Reply:
The test wasn't actually failing before but I updated it to a version where it does fail (the offsets need to be created inside the graph)

Reply:
Hmmm interesting that it was working before - I'm playing around w/ something similar to the example, and ran into the error you described.
also, with the "whether a tensor is a tracing tensor" changes, it passes but still fails when grad is enabled

Reply:
```
from torch.nested._internal.nested_tensor import NestedTensor
import torch

def fn(values, offsets):
  nt =  torch.nested.nested_tensor_from_jagged(values, offsets)
  return nt.cos()

device="cuda"
values = torch.rand((10, 8), device=device)
offsets = torch.tensor([0, 1, 3, 6, 10], device=device)

out = torch.compile(fn, fullgraph=True, backend="aot_eager")(values, offsets)
```

Reply:
From https://github.com/pytorch/pytorch/pull/124624, to fix the case when the values input requires grad, 
```
diff --git a/torch/nested/_internal/nested_tensor.py b/torch/nested/_internal/nested_tensor.py
index 5494633b09b..017c000d99c 100644
--- a/torch/nested/_internal/nested_tensor.py
+++ b/torch/nested/_internal/nested_tensor.py
@@ -15,6 +15,9 @@ def get_tensor_symint(tensor, *, coeff=1):
     global _tensor_id_counter
     tensor_symint = _tensor_symint_registry.get(tensor)
     if tensor_symint is None:
+        if isinstance(tensor, torch._subclasses.functional_tensor.FunctionalTensor):
+            tensor = torch._from_functional_tensor(tensor.elem)
+            return get_tensor_symint(tensor, coeff=coeff)
         tensor_symint = torch._C._get_nested_int(_tensor_id_counter, coeff)
         _tensor_id_counter += 1
         _tensor_symint_registry[tensor] = tensor_symint
```
The main diffference between that PR and this one is that that one uses unbacked symints over non-symbolic nested int, which I think I prefer.

Reply:
ah okay, I think I see how your suggestion works! and it does seem to fix this case as well